include/realizations/catchment/Bmi_Formulation.hpp

```diff
@@ -61,7 +63,11 @@ namespace realization {
          * @param output_stream
          */
         Bmi_Formulation(std::string id, std::unique_ptr<forcing::ForcingProvider> forcing, utils::StreamHandler output_stream)
-                : Catchment_Formulation(std::move(id), std::move(forcing), output_stream) { };
+                : Catchment_Formulation(std::move(id), std::move(forcing), output_stream)
+        {
+            // Do this here, as this function also handles initializing the output string stream for formatting.
+            set_output_precision(9);
```

Comment:
Why don't we push this into the `StreamHandler` and allow all output_streams to have formatting?